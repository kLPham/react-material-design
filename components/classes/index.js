export const buttoncssClasses = {
    colored: 'mdc-button--accent',
    accent: 'mdc-button--accent',
    compact: 'mdc-button--compact',
    dense: 'mdc-button--dense',
    primary: 'mdc-button--primary',
    raised: 'mdc-button--raised',
};
export const fabcssClasses = {
    absolute: 'app-fab--absolute',
    mini: 'mdc-fab--mini',
    plain: 'mdc-fab--plain',
};
export const listcssClasses = {
    avatar: 'mdc-list--avatar-list',
    dense: 'mdc-list--dense',
};
export const cardMediaItemcssClasses = {
    1.5: 'mdc-card__media-item--1dot5x',
    2: 'mdc-card__media-item--2x',
    3: 'mdc-card__media-item--3x',
};
export const iconcssClasses = {
    primary: 'mdc-icon-toggle--primary',
    accent: 'mdc-icon-toggle--accent',
};
export const textfieldcssClasses = {
    primary: 'mdc-theme--primary-bg mdc-theme--text-primary-on-primary',
};
