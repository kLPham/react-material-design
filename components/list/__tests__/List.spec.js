import React from 'react';
import { shallow } from 'enzyme';
import List from '../index';
import { listcssClasses } from '../../classes';

describe('<List />', () => {
    it('should render with basic li', () => {
        const wrapper = shallow(
            <List>
                <li className="mdc-list-item">
                    Item One
                </li>
                <li className="mdc-list-item">
                    Item Two
                </li>
            </List>,
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('ul')).toHaveClassName('mdc-list');
        expect(wrapper.find('li')).toBePresent();
        expect(wrapper.find('li').at(1)).toHaveText('Item Two');
    });
    it('should render dense with basic li', () => {
        const type = 'dense';
        const cssClass = listcssClasses[type];
        const wrapper = shallow(
            <List type={type}>
                <li className="mdc-list-item">
                    Item One
                </li>
                <li className="mdc-list-item">
                    Item Two
                </li>
            </List>,
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('ul')).toHaveClassName('mdc-list');
        expect(wrapper).toHaveClassName(cssClass);
        expect(wrapper.find('li')).toBePresent();
        expect(wrapper.find('li').at(1)).toHaveText('Item Two');
    });
    it('should render avatar with avatar li', () => {
        const type = 'avatar';
        const cssClass = listcssClasses[type];
        const wrapper = shallow(
            <List type={type}>
                <li className="mdc-list-item">
                    <img
                        className="mdc-list-item__start-detail" src="/users/1/profile_pic.png"
                        width="56" height="56" alt="Picture of Janet Perkins"
                    />
                        Janet Perkins
                </li>
                <li className="mdc-list-item">
                    <img
                        className="mdc-list-item__start-detail" src="/users/2/profile_pic.png"
                        width="56" height="56" alt="Picture of Mary Johnson"
                    />
                        Mary Johnson
                </li>
                <li className="mdc-list-item">
                    <img
                        className="mdc-list-item__start-detail" src="/users/3/profile_pic.png"
                        width="56" height="56" alt="Picture of Peter Carlsson"
                    />
                        Peter Carlsson
                </li>
            </List>
        ,
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('ul')).toHaveClassName('mdc-list');
        expect(wrapper).toHaveClassName(cssClass);
        expect(wrapper.find('li')).toBePresent();
        expect(wrapper.find('li').at(1)).toHaveText('Mary Johnson');
    });
});
