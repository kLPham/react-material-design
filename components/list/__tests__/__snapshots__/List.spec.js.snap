exports[`<List /> should render avatar with avatar li 1`] = `
<ul
  className="mdc-list mdc-list--avatar-list">
  <li
    className="mdc-list-item">
    <img
      alt="Picture of Janet Perkins"
      className="mdc-list-item__start-detail"
      height="56"
      src="/users/1/profile_pic.png"
      width="56" />
    Janet Perkins
  </li>
  <li
    className="mdc-list-item">
    <img
      alt="Picture of Mary Johnson"
      className="mdc-list-item__start-detail"
      height="56"
      src="/users/2/profile_pic.png"
      width="56" />
    Mary Johnson
  </li>
  <li
    className="mdc-list-item">
    <img
      alt="Picture of Peter Carlsson"
      className="mdc-list-item__start-detail"
      height="56"
      src="/users/3/profile_pic.png"
      width="56" />
    Peter Carlsson
  </li>
</ul>
`;

exports[`<List /> should render dense with basic li 1`] = `
<ul
  className="mdc-list mdc-list--dense">
  <li
    className="mdc-list-item">
    Item One
  </li>
  <li
    className="mdc-list-item">
    Item Two
  </li>
</ul>
`;

exports[`<List /> should render with basic li 1`] = `
<ul
  className="mdc-list">
  <li
    className="mdc-list-item">
    Item One
  </li>
  <li
    className="mdc-list-item">
    Item Two
  </li>
</ul>
`;
